Sources/AudioKit/Nodes/Playback/AudioPlayer/AudioPlayer.swift

```diff
@@ -284,3 +323,42 @@ extension AudioPlayer {
         schedule(at: when)
     }
 }
+
+// Extension for seeking to a specific AVAudioTime
+extension AudioPlayer {
+    /// Play the audio player from a specific time
+    /// - Parameters:
+    ///   - time: The time at which the player will seek to
+    public func seek(time: Double) {
```

Comment:
This is great.

One comment/suggestion: The way I read this, the seek function has a side affect of always playing. If the song was paused or stopped, seeking will play the song at the specified position. 

Personally with the way this function performs, I'd name it "seekAndPlay", which will fully describe what this function is doing. 

Question: Is it possible to make this function state aware? I.E. seek to the specified position without playing only when paused, so the next call to player.play() would start at the given position? This way the function caller won't need to hold onto the desired seek time until they are ready to play.

Reply:
that isn't really a seek, as jeff says, that is a play. Generally a seek would just set the player up at that time, similar to AVPlayer.seek: https://developer.apple.com/documentation/avfoundation/avplayer/1385953-seek

Honestly, these methods are still just replacing what was already made in AudioKit 4.

Reply:
if you do want a seek, rather than calling node.play() call 

`node.prepare(withFrameCount frameCount: AVAudioFrameCount)`